src/ServiceControl/Infrastructure/CriticalErrorCustomCheck.cs

```diff
@@ -0,0 +1,35 @@
+namespace ServiceControl.Operations
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.CustomChecks;
+
+    class CriticalErrorCustomCheck : CustomCheck
+    {
+        static volatile string recentFailure;
+
+        public CriticalErrorCustomCheck() 
```

Comment:
Do we need both this custom check and [this one](https://github.com/Particular/ServiceControl/pull/1776/files#diff-29ed3273db53178460f61089f4fb8a84R11)

They seem to be trying to accomplish the same thing.

Reply:
This one would be signalled if there is something wrong with the main endpoint so a slightly different purpose. The error ingestion might be signalled e.g. when we have a bug that causes each error ingestion to fail. In that case this check would be OK.